app/keepers/keepers.go

```diff
@@ -421,7 +421,7 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
 		appKeepers.keys[tokenfactorytypes.StoreKey],
 		appKeepers.GetSubspace(tokenfactorytypes.ModuleName),
 		appKeepers.AccountKeeper,
-		appKeepers.BankKeeper.WithMintCoinsRestriction(tokenfactorytypes.NewTokenFactoryDenomMintCoinsRestriction()),
```

Comment:
If this is removed, where is the restriction being applied? 

Isn't it better to modify WithMintCoinRestriction to work on pointers so the bank keeper is stays the same but the restriction is applied?

Reply:
That makes sense.

I guess the problem then is that the hooks aren't setup at this point. We end up with two bank keepers and only one gets the hooks. An idea would be to:

 * Keep a pointer to the new bank keeper that is passed to the tokenfactory: `appKeepers.TokenFactoryBankKeeper` and add the bank hook to that as well in `app.SetupHooks()`.
 
 Would have to try it to see if it works though, and we'd end up with repetition (two instances of the same hook) that may lead to other bugs in the future.

Reply:
> Would have to try it to see if it works though, and we'd end up with repetition (two instances of the same hook) that may lead to other bugs in the future.

Yes, similar to my thought process, where it technically is possible, but seems to increase technical complexity